lib/util/getFilesInFolderPerExtension.js

```diff
@@ -1,17 +1,19 @@
-const fs = require('fs');
+const { existsSync, lstatSync, readdirSync } = require('fs');
 const { format } = require('util');
-const { extname, join } = require('path');
+const { extname, join, posix, dirname } = require('path');
 
 module.exports = {
     getFilesInFolderPerExtension: (path, extension) => {
-        if (!fs.existsSync(path)) {
+        if (!existsSync(path)) {
             throw Error(format("%s it' not a file", path));
-        } else if (fs.lstatSync(path).isFile()) {
-            return [path];
+        } else if (lstatSync(path).isFile()) {
+            return [join(dirname(path), posix.basename(path))];
         }
 
-        const files = fs.readdirSync(path).filter((file) => extname(file) === format('.%s', extension));
-        files.forEach((item, index, arr) => (arr[index] = join(path, item)));
+        const files = readdirSync(path).filter(file => extname(file) === format('.%s', extension));
+        files.forEach((item, index, arr) => {
+            arr[index] = join(path, item);
+        });
```

Comment:
Immutability - assign `files.map(item => join(path, item))` to a new array and return it.

Reply:
Solved